docs/api.md

```diff
@@ -0,0 +1,253 @@
+# soundslike - API documentation
+Learn how to use the soundslike API!
+
+First. for the sake of this, im going to use soundslike.com as the url, however localhost:8888 might be the url if you're using docker 
+
+## Section 1. Authentication 
+
+### Getting auth token from api
```

Comment:
I think we should swap the header of the subsection to be `GET /api/token` and have the text of the description be something like "This gets the auth token from api." And use that pattern for this document.